python/ppml/src/bigdl/ppml/ppml_context.py

```diff
@@ -0,0 +1,102 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from bigdl.ppml.api import *
+import argparse
+
+
+class PPMLContext(JavaValue):
+    def __init__(self, jvalue=None, *args):
+        self.bigdl_type = "float"
+        super().__init__(jvalue, self.bigdl_type, *args)
+
+    def load_keys(self, primary_key_path, data_key_path):
+        callBigDlFunc(self.bigdl_type, "loadKeys", self.value, primary_key_path, data_key_path)
+
+    def read(self, crypto_mode):
+        df_reader = callBigDlFunc(self.bigdl_type, "read", self.value, crypto_mode)
+        return EncryptedDataFrameReader(self.bigdl_type, df_reader)
+
+    def write(self, dataframe, crypto_mode):
+        df_writer = callBigDlFunc(self.bigdl_type, "write", self.value, dataframe, crypto_mode)
+        return EncryptedDataFrameWriter(self.bigdl_type, df_writer)
+
+
+class EncryptedDataFrameReader:
+    def __init__(self, bigdl_type, df_reader):
+        self.bigdl_type = bigdl_type
+        self.df_reader = df_reader
+
+    def option(self, key, value):
+        self.df_reader = callBigDlFunc(self.bigdl_type, "option", self.df_reader, key, value)
+        return self
+
+    def csv(self, path):
+        return callBigDlFunc(self.bigdl_type, "csv", self.df_reader, path)
+
+
+class EncryptedDataFrameWriter:
+    support_mode = {"overwrite", "append", "ignore", "error", "errorifexists"}
+
+    def __init__(self, bigdl_type, df_writer):
+        self.bigdl_type = bigdl_type
+        self.df_writer = df_writer
+
+    def option(self, key, value):
+        self.df_writer = callBigDlFunc(self.bigdl_type, "option", self.df_writer, key, value)
+        return self
+
+    def mode(self, mode):
+        if mode not in EncryptedDataFrameWriter.support_mode:
+            raise ValueError("Unknown save mode: " + mode + "." +
+                             "Accepted save modes are 'overwrite', 'append', 'ignore', 'error', 'errorifexists'.")
+        self.df_writer = callBigDlFunc(self.bigdl_type, "mode", self.df_writer, mode)
+        return self
+
+    def csv(self, path):
+        return callBigDlFunc(self.bigdl_type, "csv", self.df_writer, path)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
```

Comment:
wrap this to a class.

Reply:
remove the usage example in `ppml_context.py` , and add a new file `simple_query_example.py`